operators/src/processing/rgb.rs

```diff
@@ -0,0 +1,615 @@
+use snafu::{ensure, Snafu};
+
+/// The `Rgb` operator combines three raster sources into a single raster output.
+/// The output it of type `U32` and contains the red, green and blue values as the first, second and third byte.
+/// The forth byte (alpha) is always 255.
+pub type Rgb = Operator<RgbParams, RgbSources>;
+
+/// Parameters for the `Rgb` operator.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, Copy)]
+#[serde(rename_all = "camelCase")]
+pub struct RgbParams {
+    /// The minimum value for the red channel.
+    pub red_min: f64,
+    /// The maximum value for the red channel.
+    pub red_max: f64,
+    /// A scaling factor for the red channel between 0 and 1.
+    #[serde(default = "num_traits::One::one")]
+    pub red_scale: f64,
+
+    /// The minimum value for the red channel.
+    pub green_min: f64,
+    /// The maximum value for the red channel.
+    pub green_max: f64,
+    /// A scaling factor for the green channel between 0 and 1.
+    #[serde(default = "num_traits::One::one")]
+    pub green_scale: f64,
+
+    /// The minimum value for the red channel.
+    pub blue_min: f64,
+    /// The maximum value for the red channel.
+    pub blue_max: f64,
+    /// A scaling factor for the blue channel between 0 and 1.
+    #[serde(default = "num_traits::One::one")]
+    pub blue_scale: f64,
+}
+
+impl RgbParams {
+    fn check_valid_user_input(&self) -> Result<()> {
+        ensure!(
+            self.red_min < self.red_max,
+            error::MinGreaterThanMax {
+                color: "red",
+                min: self.red_min,
+                max: self.red_max,
+            }
+        );
+        ensure!(
+            self.green_min < self.green_max,
+            error::MinGreaterThanMax {
+                color: "green",
+                min: self.green_min,
+                max: self.green_max,
+            }
+        );
+        ensure!(
+            self.blue_min < self.blue_max,
+            error::MinGreaterThanMax {
+                color: "blue",
+                min: self.blue_min,
+                max: self.blue_max,
+            }
+        );
+
+        ensure!(
+            (0.0..=1.0).contains(&self.red_scale)
+                && (0.0..=1.0).contains(&self.green_scale)
+                && (0.0..=1.0).contains(&self.blue_scale),
+            error::RgbScaleOutOfBounds {
+                red: self.red_scale,
+                green: self.green_scale,
+                blue: self.blue_scale
+            }
+        );
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RgbSources {
+    red: Box<dyn RasterOperator>,
+    green: Box<dyn RasterOperator>,
+    blue: Box<dyn RasterOperator>,
+}
+
+impl OperatorData for RgbSources {
+    fn data_names_collect(&self, data_names: &mut Vec<NamedData>) {
+        for source in self.iter() {
+            source.data_names_collect(data_names);
+        }
+    }
+}
+
+impl RgbSources {
+    pub fn new_a(
```

Comment:
why is it called "new_a"? Is there a "b"?

Reply:
[d397cc5](https://github.com/geo-engine/geoengine/pull/807/commits/d397cc52ce31074c401214b88537b8fa5a42fb26)